twine/auth.py

```diff
@@ -53,13 +59,41 @@ def username(self) -> Optional[str]:
     @property
     @functools.lru_cache()
     def password(self) -> Optional[str]:
+        if self.oidc:
```

Comment:
We should probably gate this on whether it's enabled and whether the repository is PyPI as well

Reply:
It looks like trusted publishing is designed to work with other indexes - going by the URL to look up what audience it expects. But I imagine it's only [Test]PyPI that currently supports it. How do you weigh up nicer error messages versus the need to make more changes in the future?

Reply:
To the best of my knowledge most other indexes that folks would use Twine with won't support this any time soon